rmc/utils/mpc.py

```diff
@@ -433,51 +433,78 @@ def _run_glm(
         min_single_aic_var,
     )
 
-    # List possible variable combinations for Joint Regression
-    comb_var = []
-    for k in range(2,6):
-        comb_var += [x for x in combinations(all_var,k)]
+    # List possible variable combinations for joint regression
+    var_combs = []
+    for k in range(2, len(all_var) + 1):
+        var_combs += [x for x in combinations(all_var, k)]
 
-    logger.info("Running joint (additive interactions only) regression...")
     add_model_res = {}
```

Comment:
General comment: Can we extract out (functionalize)the code in `run_regressions` that is duplicated over the single, additive, and multiplicative code blocks?